lib/logger/__tests__/logger.test.ts

```diff
@@ -80,6 +94,33 @@ describe('shopify.logger', () => {
         expect.stringContaining('error message'),
       );
     });
+
+    /* eslint-disable no-process-env */
+    it('can log deprecations for older versions outside of CI', async () => {
+      delete process.env.CI;
+
+      const testMessage = 'This is a test';
+      await shopify.logger.deprecate('1.0.0', testMessage, {
+        testContext: true,
+      });
+
+      expect(shopify.config.logger.log).toHaveBeenCalledWith(
+        LogSeverity.Warning,
+        expect.stringContaining('[Deprecation warning]'),
+      );
+    });
+
+    it('fails to log deprecations for older versions on CI', async () => {
```

Comment:
If it is past the deprecation schedule, shouldn't the functions be removed from the library?

I'd think the failing assertions would be in context of the functions that are deprecated.

```js
function coolFunctionBro() {
  shopify.logger("2.0.0")
}

# test
test("coolFunctionBro", () => {
  #expect shopify.logger.deprecate called
  # compare within deprecation schedule as a separate assertion
});
```